server/svix-server/src/queue/mod.rs

```diff
@@ -6,13 +6,38 @@ use serde::{Deserialize, Serialize};
 use svix_ksuid::*;
 
 use crate::{
+    cfg::{Configuration, QueueType},
     core::types::{ApplicationId, EndpointId, MessageAttemptTriggerType, MessageId},
     error::Result,
 };
 
 pub mod memory;
 pub mod redis;
 
+pub async fn new_pair(
+    cfg: Configuration,
```

Comment:
One comment (I know it's already merged): this is not something we like doing and I quite dislike it.

We don't pass large structures like this to functions if we can avoid it (not sure we can in this case) because it makes things painful to test and painful to reason about.